tls/extensions/s2n_extension_type.c

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <s2n.h>
+
+#include "error/s2n_errno.h"
+#include "tls/extensions/s2n_extension_type.h"
+#include "tls/s2n_client_extensions.h"
+#include "tls/s2n_tls_parameters.h"
+#include "utils/s2n_safety.h"
+
+#define TLS_EXTENSION_DATA_LENGTH_BYTES 2
+
+/* Because there are 65536 possible extension IANAs, we will only
+ * put the lowest (and most common) in a lookup table to conserve space. */
+#define S2N_MAX_INDEXED_EXTENSION_IANA 60
+
+const uint16_t s2n_supported_extensions[] = {
+    TLS_EXTENSION_RENEGOTIATION_INFO,
+    TLS_EXTENSION_PQ_KEM_PARAMETERS,
+    TLS_EXTENSION_SERVER_NAME,
+    TLS_EXTENSION_MAX_FRAG_LEN,
+    TLS_EXTENSION_STATUS_REQUEST,
+    TLS_EXTENSION_SUPPORTED_GROUPS,
+    TLS_EXTENSION_EC_POINT_FORMATS,
+    TLS_EXTENSION_SIGNATURE_ALGORITHMS,
+    TLS_EXTENSION_ALPN,
+    TLS_EXTENSION_SCT_LIST,
+    TLS_EXTENSION_SESSION_TICKET,
+    TLS_EXTENSION_SUPPORTED_VERSIONS,
+    TLS_EXTENSION_KEY_SHARE,
+};
+const s2n_extension_type_id s2n_supported_extensions_count = sizeof(s2n_supported_extensions) / sizeof(uint16_t);
+const s2n_extension_type_id s2n_unsupported_extension = sizeof(s2n_supported_extensions) / sizeof(uint16_t);
+
+s2n_extension_type_id s2n_extension_ianas_to_ids[S2N_MAX_INDEXED_EXTENSION_IANA];
+
+int s2n_extension_type_init()
+{
+    /* Initialize to s2n_unsupported_extension */
+    for (int i = 0; i < S2N_MAX_INDEXED_EXTENSION_IANA; i++) {
```

Comment:
is `int` the right type for a loop variable here?

Reply:
What type were you thinking? I could make it a uint8_t, but int seems much more common in the codebase. It ranges from 0 -> 59.